### PR TITLE
Add hyperparameter tuning example.

### PR DESCRIPTION
**Pull request by coledie**
- [x] Forked repo is synced with upstream -> github shows no code delta outside of the desired.
    https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork
- [x] Tests are passing? https://github.com/microsoft/DeepGNN/blob/main/CONTRIBUTING.md#run-tests
- [x] Changelog and documentation updated.
- [x] PR is labeled using the label menu on the right side.

New Behavior
----------------
Add simple ray hyperparameter tuning example.

### CODE REVIEW

File: docs/advanced/meta_opt.rst [hunk 1]
```diff
@@ -0,0 +1,251 @@
+*****************************************
+Hyperparameter Optimization with Ray Tune
+*****************************************
+
+In this guide we build on top of the Ray usage example with a Ray Tune example at the bottom.
+The following code block is from `node_class example <https://github.com/microsoft/DeepGNN/blob/main/docs/torch/node_class.rst>`_, see this example for more details.
```
Comment (mortid0) [inline docs/advanced/meta_opt.rst:6]: Please, reference a doc from readthedocs, not a github page.

File: docs/advanced/meta_opt.rst [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    ...         verbose=0,
+    ...     ),
+    ... )
+    >>> analysis = tuner.fit()
+
+    >>> analysis.get_best_result(metric="accuracy", mode="max")
+    Result(metrics={'accuracy': ..., 'experiment_tag': '..._learning_rate=...,n_epochs=...'}, ...)
+
+    >>> analysis.get_dataframe()
+       accuracy  ...
```
Comment (mortid0) [inline docs/advanced/meta_opt.rst:250]: Please, check actual numbers in the data frame.

File: docs/advanced/meta_opt.rst [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    ...             "n_epochs": n_epochs,
+    ...             "learning_rate": learning_rate,
+    ...         },
+    ...         run_config=RunConfig(verbose=0),
+    ...         scaling_config=ScalingConfig(num_workers=1, use_gpu=False),
+    ...     )
+    ...     result = trainer.fit()
+    ...     return result
+
+This training function just wraps this objective function for use in the tuner.
```
Comment (mortid0) [inline docs/advanced/meta_opt.rst:215]: nit: remove `just`

File: docs/advanced/meta_opt.rst [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+.. code-block:: python
+
+    >>> def training_function(config):
+    ...    learning_rate, n_epochs = config["learning_rate"], config["n_epochs"]
+    ...    for step in range(1):
```
Comment (mortid0) [inline docs/advanced/meta_opt.rst:221]: Do we even need this loop?

File: docs/advanced/meta_opt.rst [hunk 5]
```diff
@@ -0,0 +1,251 @@
+*****************************************
+Hyperparameter Optimization with Ray Tune
+*****************************************
+
+In this guide we build on top of the Ray usage example with a Ray Tune example at the bottom.
+The following code block is from `node_class example </torch/node_class.rst>`_, see this example for more details.
```
Comment (mortid0) [inline docs/advanced/meta_opt.rst:6]: that's a broken link.

File: docs/advanced/meta_opt.rst [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+This training function wraps this objective function for use in the tuner.
+
+.. code-block:: python
+
+    >>> def training_function(config):
+    ...    results = objective(config["learning_rate"], config["n_epochs"])
+    ...    tune.report(accuracy=results.metrics["metric"])
+
+Finally we define and make use of the tuner. We use the hyperparameters learning_rate and
```
Comment (mortid0) [inline docs/advanced/meta_opt.rst:223]: nit: highlight `learning_rate` and `n_epochs`

**Review by mortid0 — APPROVED** [review]
